ssawg_trending_scraper.py

```diff
@@ -43,112 +40,59 @@
 
 URL_ASPECT = "https://cxc.cfa.harvard.edu/mta/ASPECT"
 
-trending_pages = {
-    "Reports": [
-        "acq_stat_reports",
-        "gui_stat_reports",
-        "periscope_drift_reports", # periscope page requires a username/password
-    ],
-    "Perigee": [
-        "perigee_health_plots",
-    ],
-    "Generic": [
-        "kalman_watch", 
-        "obc_rate_noise/trending",
-        "fid_drift",
-        "aimpoint_mon",
-        "celmon",
-        "vv_rms",
-        "attitude_error_mon",
-        "fss_check3",
-    ]
-}
+
+def get_elements(soup, element):
+    """
+    Grab various elements from page.
+    """
+    return list(soup.find_all(element))
+
+
+def get_images(soup, image, url):
+    """
+    Find all pngs and gifs.
+    """
+    images = {}
+    for img in soup.find_all(image):
+        if img['src'].endswith('.png') or img['src'].endswith("gif"):
+            # look for all pngs and gifs
+            new_image_url = f'<img src = "{url}{img["src"]}">'
+            images[img['src']] = new_image_url
+    return images
+
+
+def get_tables(soup, tbl, url):
+    """
+    Find all tables.
+    """
+    tables = [table for table in soup.find_all(tbl)]
+    new_tables = [str(table) for table in tables]
+    for index, table in enumerate(new_tables):
+        # replace truncated img src url calls with full url calls;
+        # this allows the script to be run/tested outside network
+        new_tables[index] = re.sub('src="', f'src="{url}/', str(table))
+    return new_tables
+
 
 # ---------------------------------
 # Establish HTML info for each page
 # ---------------------------------
-
-class Page():
+class BasePage():
     """
-    Establish page, URL, and HTML information.
+    Base class for establishing page, URL, and HTML information.
     """
+    auth = None
+
     def __init__(self, page):
         self.page = page
-    
-        def get_proper_urls_reports(page):
```

Comment:
It's not a normal idiom to have all these functions inside the scope of `__init__`.  They should be normal object methods at the same level as `__init__`.